packages/framework/tests/Unit/Component/Money/MoneyTest.php

```diff
@@ -0,0 +1,582 @@
+     */
+    public function testRound(string $amount, int $scale, string $expectedResult): void
+    {
+        $money = Money::fromString($amount);
+
+        $result = $money->round($scale);
+
+        $this->assertEquals($expectedResult, $result->toString());
+    }
+
+    /**
+     * @dataProvider roundProvider
+     * @param string $amount
+     * @param int $scale
+     */
+    public function testRoundingIsSameAsInPhp(string $amount, int $scale): void
+    {
+        $money = Money::fromString($amount);
+
+        $result = $money->round($scale);
+        $expectedResult = round($amount, $scale);
+
+        $this->assertEquals($expectedResult, $result->toString());
+    }
+
+    /**
+     * @return \Iterator
+     */
+    public function roundProvider(): Iterator
+    {
+        yield ['0.4', 0, '0'];
+        yield ['-0.4', 0, '-0'];
+        yield ['0.5', 0, '1'];
+        yield ['-0.5', 0, '-1'];
+        yield ['1.5', 0, '2'];
+        yield ['-1.5', 0, '-2'];
+        yield ['0.000000', 2, '0.00'];
+        yield ['1.000000', 2, '1.00'];
+        yield ['0.00', 2, '0.00'];
+        yield ['1.00', 2, '1.00'];
+        yield ['0.0', 2, '0.0'];
+        yield ['1.0', 2, '1.0'];
+        yield ['0', 2, '0'];
+        yield ['1', 2, '1'];
+        yield ['1.999', 3, '1.999'];
+        yield ['1.999', 2, '2.00'];
+    }
+
+    /**
+     * @dataProvider compareProvider
+     * @param string $a
+     * @param string $b
+     * @param int $expectedResult
+     */
+    public function testCompare(string $a, string $b, int $expectedResult): void
+    {
+        $moneyA = Money::fromString($a);
+        $moneyB = Money::fromString($b);
+
+        $result = $moneyA->compare($moneyB);
+
+        $this->assertEquals($expectedResult, $result);
+    }
+
+    /**
+     * @return \Iterator
+     */
+    public function compareProvider(): Iterator
+    {
+        yield ['0', '0', 0];
+        yield ['1', '1', 0];
+        yield ['0.0', '0.0', 0];
+        yield ['0', '0.0', 0];
+        yield ['0.0', '0', 0];
+        yield ['0', '0.000', 0];
+        yield ['0.000', '0', 0];
+        yield ['0.0001', '0.0001000', 0];
+        yield ['0', '-0', 0];
+        yield ['1', '0', 1];
+        yield ['0', '1', -1];
+        yield ['1', '-1', 1];
+        yield ['-1', '1', -1];
+        yield ['0.000001', '0', 1];
+        yield ['0', '0.000001', -1];
+    }
+
+    /**
+     * @dataProvider equalsProvider
+     * @param string $a
+     * @param string $b
+     * @param bool $expectedResult
+     */
+    public function testEquals(string $a, string $b, bool $expectedResult): void
+    {
+        $moneyA = Money::fromString($a);
+        $moneyB = Money::fromString($b);
+
+        $result = $moneyA->equals($moneyB);
+
+        $this->assertEquals($expectedResult, $result);
+    }
+
+    /**
+     * @return \Iterator
+     */
+    public function equalsProvider(): Iterator
+    {
+        foreach ($this->compareProvider() as $dataset) {
```

Comment:
Please no. Please duplicate it here in form `a, b, expected`. This is unreadable and difficult to understand

Reply:
The same is also valid for tests below

Reply:
Fixed in 6cd43983e